changelogs/fragments/1152-lambda_info-bugfix-all-lambdas.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - lambda_info - fix bug that forces query=config when getting info for all lambdas (https://github.com/ansible-collections/community.aws/pull/1152).
```

Comment:
From experience with other AWS APIs I suspect this may have been a deliberate decision rather than a "bug".

Reply:
This also doesn't really express that the new **default** when getting the info for all lambdas is query=all

Reply:
@tremble While it may have been intentional, I'd still classify this as a bug since as a user I would expect all info returned when using `query=all`. As is, the module ignores what was input and uses `query=config`.

I updated the logic for setting `query`s default value if omitted to match your suggestion